### PR TITLE
Allow label elements

### PR DESCRIPTION
**Pull request by nol13**
Allow labelExtractor to return an element instead of a string, to allow for showing custom icons or conditional styling and such. (my first time updating flow types but think those are right)

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -22,7 +22,7 @@ import TagInput from 'react-native-tag-input';
...
 | value                | (Required) An array of tags, which can be any type, as long as labelExtractor below can extract a string from it                                                                                                                   |
 | onChange             | (Required) A handler to be called when array of tags change. The parent should update the value prop when this is called if they want to enable removal of tags                                                                    |
-| labelExtractor       | (Required) Function to extract string value for label from item                                                                                                                                                                    |
+| labelExtractor       | (Required) Function to extract string value for label from item. May also return an element to be shown in place of text.                                                                                                                                                                    |
```
Comment (Ashoat) [inline README.md]: Can you update this to say:

> "May also return an element to be shown in place of text, in which case the `tagTextColor` and `tagTextStyle` props will be ignored."

File: index.js [hunk 2]
```diff
@@ -405,14 +405,15 @@ class Tag extends React.PureComponent<TagProps> {
           this.props.tagContainerStyle,
         ]}
       >
-        <Text style={[
+        {React.isValidElement(this.props.label) ? this.props.label :
```
Comment (Ashoat) [inline index.js]: I find that using ternary syntax inline in JSX feels pretty messy. Can you extract this condition to the start of the `render` function?

**Review by Ashoat — CHANGES_REQUESTED** [review]
Thanks for this PR, @nol13. I think you got the types right! Did you try running `flow` to make sure everything typechecks?

I have some requested changes inline. Other than that, for my curiosity's sake - what do you need this new prop for?

**Comment by nol13** [comment]
I did run flow. Am using passing a component as a way to do conditional styling at the moment. We want some invalid tags to still get made into tags, but have their text show up red, while valid tags would still be gray.

Actually in real app am currently returning Text component from labelExtractor anyway, and ignoring the PropType warnings, but figured this makes it a little more flexible.

**Comment by Ashoat** [comment]
Published in `react-native-tag-input@0.0.21`. Thanks again!

**Pull request merged**